gluten-ut/spark35/src/test/scala/org/apache/spark/sql/GlutenJsonFunctionsSuite.scala

```diff
@@ -16,4 +16,88 @@
+    runTest(json, "$.store.book[*]", null) // not supported in velox
+    runTest(json, "$.store.book[0].category", "reference")
+    runTest(json, "$.store.book[*].category", null) // not supported in velox
+    runTest(json, "$.store.book[*].reader[0].age", null) // not supported in velox
+    runTest(json, "$.store.book[*].reader[*].age", null) // not supported in velox
+    runTest(json, "$.store.basket[0][1]", "2")
+    runTest(json, "$.store.basket[*]", null) // not supported in velox
+    runTest(json, "$.store.basket[*][0]", null) // not supported in velox
+    runTest(json, "$.store.basket[0][*]", null) // not supported in velox
+    runTest(json, "$.store.basket[*][*]", null) // not supported in velox
+    runTest(json, "$.store.basket[0][2].b", "y")
+    runTest(json, "$.store.basket[0][*].b", null) // not supported in velox
+    runTest(json, "$.non_exist_key", null)
+    runTest(json, "$.store.book[10]", null)
+    runTest(json, "$.store.book[0].non_exist_key", null)
+    runTest(json, "$.store.basket[*].non_exist_key", null) // not supported in velox
+    runTest(json, "$.store.basket[0][*].non_exist_key", null) // not supported in velox
+    runTest(json, "$.store.basket[*][*].non_exist_key", null) // not supported in velox
+    runTest(json, "$.zip code", "94025")
+    runTest(json, "$.fb:testid", "1234")
+    runTest("{\"a\":\"b\nc\"}", "$.a", null) // not supported in velox
+
+    // Test root array
+    runTest("[1,2,3]", "$[0]", "1")
+    runTest("[1,2,3]", "$.[0]", null) // Not supported in spark and velox
+    runTest("[1,2,3]", "$.[1]", null) // Not supported in spark and velox
+    runTest("[1,2,3]", "$[1]", "2")
+
+    runTest("[1,2,3]", "$[3]", null)
+    runTest("[1,2,3]", "$.[*]", null) // Not supported in spark and velox
+    runTest("[1,2,3]", "$[*]", null) // not supported in velox
+    runTest("[1,2,3]", "$", null) // not supported in velox
+    runTest("[{\"k1\":\"v1\"},{\"k2\":\"v2\"},{\"k3\":\"v3\"}]", "$[2]", "{\"k3\":\"v3\"}")
+    runTest("[{\"k1\":\"v1\"},{\"k2\":\"v2\"},{\"k3\":\"v3\"}]", "$[2].k3", "v3")
+    runTest("[{\"k1\":[{\"k11\":[1,2,3]}]}]", "$[0].k1[0].k11[1]", "2")
+    runTest("[{\"k1\":[{\"k11\":[1,2,3]}]}]", "$[0].k1[0].k11", "[1,2,3]")
+    runTest("[{\"k1\":[{\"k11\":[1,2,3]}]}]", "$[0].k1[0]", "{\"k11\":[1,2,3]}")
+    runTest("[{\"k1\":[{\"k11\":[1,2,3]}]}]", "$[0].k1", "[{\"k11\":[1,2,3]}]")
+    runTest("[{\"k1\":[{\"k11\":[1,2,3]}]}]", "$[0]", "{\"k1\":[{\"k11\":[1,2,3]}]}")
+    runTest("[[1,2,3],[4,5,6],[7,8,9]]", "$[1]", "[4,5,6]")
+    runTest("[[1,2,3],[4,5,6],[7,8,9]]", "$[1][0]", "4")
+    runTest("[\"a\",\"b\"]", "$[1]", "b")
+    runTest("[[\"a\",\"b\"]]", "$[0][1]", "b")
+
+    runTest("[1,2,3]", "[0]", "1")
+    // runTest("[1,2,3]", "$0", null) crashes in velox
+    runTest("[1,2,3]", "0", null)
+    runTest("[1,2,3]", "$.", null)
+
+    // runTest("[1,2,3]", "$", null) crashes in velox
+    // runTest("{\"a\":4}", "$", null) crashes in velox
```

Comment:
I just created one issue to track the above two cases: https://github.com/apache/incubator-gluten/issues/5504.

According to Spark code, I note input json is the expected result for these two tests. Please keep them unchanged (still commented).

Reply:
Made the change